### PR TITLE
kie-issues#964: Adapt kie-tools Jenkins staging/release jobs to build and deploy Quarkus Dev UI modules to Nexus/Maven Central 

### LINKED ISSUE #964: Adapt `kie-tools` Jenkins staging/release jobs to build and deploy Quarkus Dev UI modules to Nexus/Maven Central

### PR DESCRIPTION
**Pull request by rodrigonull**
Closes: https://github.com/apache/incubator-kie-issues/issues/964

In this PR we add steps to build, test and deploy DevUI java packages (jbpm-quarkus-devui  e sonataflow-quarkus-devui) to Apache maven repository.

Only the daily dev and staging pipelines are affected by this change. I talked with @porcelli and the release promotion of these packages will be a manual step for the first Apache release.

Depends on: https://github.com/apache/incubator-kie-tools/pull/2270

### CODE REVIEW

**Review by yesamer — COMMENTED** [review]
@rodrigonull You need to update the lock file.
Just type `pnpm bootstrap` in the project root dir.

**Review by thiagoelg — APPROVED** [review]

**Review by baldimir — APPROVED** [review]

**Comment by yesamer** [comment]
@rodrigonull Now you need to update `graph.dot` too, launch again `pnpm bootstrap` :)

**Comment by tiagobento** [comment]
As we discussed, the relevant Maven packages will now have the `build:prod` command prepared to deploy when 

```bash
export KIE_TOOLS_BUILD__mavenDeploySkip=false
````

cc @jstastny-cz

**Comment by rodrigonull** [comment]
Once the [PR#2262 ](https://github.com/apache/incubator-kie-tools/pull/2262) is merged, I will rebase this PR and make the necessary adjustments.

**Review by porcelli — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]

**Review by thiagoelg — APPROVED** [review]

**Review by porcelli — APPROVED** [review]

File: .ci/jenkins/Jenkinsfile.daily-dev-publish [hunk 1]
```diff
@@ -530,6 +530,27 @@ pipeline {
...
+
+        stage('Build and Deploy (jbpm-quarkus-devui and sonataflow-quarkus-devui)') {
+            steps {
+                dir('kie-tools') {
+                    script {
+                        withCredentials([usernamePassword(credentialsId: "${pipelineVars.mavenDeployRepositoryCredentialsId}", usernameVariable: 'REPOSITORY_USER', passwordVariable: 'REPOSITORY_TOKEN')]) {
+                            configFileProvider([configFile(fileId: "${pipelineVars.mavenSettingsConfigFileId}", variable: 'MAVEN_SETTINGS_FILE')]) {
+                                sh """#!/bin/bash -el
+                                export KIE_TOOLS_BUILD__mavenDeploySkip=false
+                                pnpm update-version-to 0.0.0-SNAPSHOT -F jbpm-quarkus-devui... -F sonataflow-quarkus-devui...
```
Comment (tiagobento) [inline .ci/jenkins/Jenkinsfile.daily-dev-publish]: 999-SNAPSHOT for consistency

Comment (tiagobento) [reply .ci/jenkins/Jenkinsfile.daily-dev-publish]: -Drevision=999-SNAPSHOT

**Review by tiagobento — APPROVED** [review]

**Pull request merged**